src/Common/Scheduler/ResourceRequest.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <boost/intrusive/list.hpp>
```

Comment:
I hate having this include here, because the header is heavily used (and intrusive list looks like an implementation detail). But for some reason intrusive list custom hook with size tracking is not compatible with abstract classes (and we have `execute() = 0`):
```
boost/intrusive/list.hpp:138:26: error: field type 'header_holder_type' (aka 'default_header_holder<Q::NT>') is an abstract class
```